src/main/java/core/basesyntax/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class UserDaoImpl implements UserDao<User> {
+
+    @Override
+    public Optional<User> get(long id) {
+        return Optional.ofNullable(Storage.users.get((int) id));
+    }
+
+    @Override
+    public List<User> getAll() {
+        return Storage.users;
+    }
+
+    @Override
+    public void save(User user) {
+        Storage.users.add(user);
+    }
+
+    @Override
+    public void update(User user, String [] params) {
```

Comment:
```suggestion
    public void update(User user) {
```
you don't need params, just pass user with already updated values